Apps/WebClient/src/NewClientApp/src/components/public/vaccine-card/VaccineCardView.vue

```diff
@@ -0,0 +1,638 @@
+                                                v$.dateOfBirth
+                                            )
+                                        "
+                                        :error-messages="
+                                            ValidationUtil.getErrorMessages(
+                                                v$.dateOfBirth
+                                            )
+                                        "
+                                        @blur="v$.dateOfBirth.$touch()"
+                                    />
+                                </div>
+                            </v-col>
+                        </v-row>
+                    </div>
+                    <div>
+                        <v-row>
+                            <v-col>
+                                <label for="dateOfVaccine"
+                                    >Date of Vaccine (Any Dose)</label
+                                >
+                                <div>
+                                    <HgDatePickerComponent
+                                        id="dateOfVaccine"
+                                        v-model="dateOfVaccine"
+                                        label="Date of Vaccine (Any Dose)"
+                                        data-testid="dateOfBirthInput"
+                                        :state="
+                                            ValidationUtil.isValid(
+                                                v$.dateOfVaccine
+                                            )
+                                        "
+                                        :error-messages="
+                                            ValidationUtil.getErrorMessages(
+                                                v$.dateOfVaccine
+                                            )
+                                        "
+                                        @blur="v$.dateOfVaccine.$touch()"
+                                    />
+                                </div>
+                            </v-col>
+                        </v-row>
+                    </div>
+                    <v-row class="pt-4 justify-content-between">
+                        <v-col cols="6">
+                            <HgButtonComponent
+                                variant="secondary"
+                                aria-label="Cancel"
+                                data-testid="btnCancel"
+                                block
+                                to="/"
+                            >
+                                Cancel
+                            </HgButtonComponent>
+                        </v-col>
+                        <v-col cols="6">
+                            <HgButtonComponent
+                                variant="primary"
+                                aria-label="Enter"
+                                :disabled="
```

Comment:
Rather set `:loading` when dealing with promises, this will both block and display a loading animation on the button, use `:disable` when monitoring validation.